### PR TITLE
Forwards the logs inside the docker containers out to slf4j

### PR DESCRIPTION
**Pull request by karlney**

### CODE REVIEW

File: src/main/kotlin/com/meltwater/docker/compose/DockerCompose.kt [hunk 1]
```diff
@@ -63,6 +76,21 @@ class DockerCompose(classPathYmlResource: String,
         return ps
     }
 
+    private fun forwardDockerLog(procLogger: ProcessWrapper) {
+        thread {
+            try {
+                val exitCode = procLogger.waitFor()
+                if (exitCode != 0) {
+                    LOGGER.error("The 'logs -f' command finished with a non 0 exit code '{}'", exitCode)
```
Comment (karlney) [inline src/main/kotlin/com/meltwater/docker/compose/DockerCompose.kt:84]: Don't know how to test this (even manually). Is it needed?

File: src/main/kotlin/com/meltwater/docker/compose/ExecUtils.kt [hunk 2]
```diff
@@ -92,7 +97,7 @@ object ExecUtils {
         }
     }
 
-    private fun collectAndLogResult(inputStream: InputStream, listener: (String) -> Unit): String {
+    private fun collectOutput(inputStream: InputStream, listener: (String) -> Unit): String {
```
Comment (karlney) [inline src/main/kotlin/com/meltwater/docker/compose/ExecUtils.kt:99]: Nothing was logged from here

File: src/main/kotlin/com/meltwater/docker/compose/ExecUtils.kt [hunk 3]
```diff
@@ -78,10 +82,11 @@ object ExecUtils {
 
             val process = pb.start()
 
-            val result = collectAndLogResult(process.inputStream, listener)
+            val result = collectOutput(process.inputStream, listener)
```
Comment (karlney) [inline src/main/kotlin/com/meltwater/docker/compose/ExecUtils.kt:82]: @bergetp should we log the result output on trace?

File: src/main/kotlin/com/meltwater/docker/compose/ProcessWrapper.kt [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+import kotlin.concurrent.thread
+
+class ProcessWrapper(val pb: ProcessBuilder, stdOutListener: (String) -> Unit, stdErrListener: (String) -> Unit){
+
+    val LOGGER: Logger = LoggerFactory.getLogger(ExecUtils::class.java)
+
+    val process: Process
+    val stdOutReader: Thread
+    val stdErrReader: Thread
+    var exitCode: Int = 0
```
Comment (karlney) [inline src/main/kotlin/com/meltwater/docker/compose/ProcessWrapper.kt]: If you ask for the exit code before termination you get 0 - a bit dangerous. Change to -1

File: src/main/kotlin/com/meltwater/docker/compose/DockerCompose.kt [hunk 5]
```diff
@@ -26,6 +28,15 @@ class DockerCompose(classPathYmlResource: String,
         LOGGER.info(line)
     }
 
+    val STDOUT_LOG_CONSUMER = { line: String ->
+        LOGGER.trace("OUT: "+line)
```
Comment (karlney) [inline src/main/kotlin/com/meltwater/docker/compose/DockerCompose.kt]: @bergetp should this be DEBUG or TRACE ??

File: src/main/kotlin/com/meltwater/docker/compose/ExecUtils.kt (hunk 3 above)
Comment (karlney) [reply src/main/kotlin/com/meltwater/docker/compose/ExecUtils.kt:82]: I''ll add it for now

File: src/main/kotlin/com/meltwater/docker/compose/ProcessWrapper.kt (hunk 4 above)
Comment (karlney) [reply src/main/kotlin/com/meltwater/docker/compose/ProcessWrapper.kt]: Fixing

File: src/main/kotlin/com/meltwater/docker/compose/DockerCompose.kt (hunk 5 above)
Comment (bergetp) [reply src/main/kotlin/com/meltwater/docker/compose/DockerCompose.kt]: DEBUG I think

File: src/main/kotlin/com/meltwater/docker/compose/DockerCompose.kt (hunk 1 above)
Comment (bergetp) [reply src/main/kotlin/com/meltwater/docker/compose/DockerCompose.kt:84]: Manually :)

**Pull request merged**